From 8f7ce559b84ee0c21de0256e6591793e4b95f411 Mon Sep 17 00:00:00 2001
From: David Redondo <kde@david-redondo.de>
Date: Tue, 14 Jul 2020 13:02:51 +0200
Subject: [PATCH] Do not remove virtual method from build

Method was deprecated in fc5738ab7c4025a87e2bbd656914b6187df7a9eb. Removing the
method when building without deprecated methods (for example the pim runner
does this) changes the vtable leading to crashes.
BUG: 423003
---
 src/abstractrunner.cpp | 2 --
 src/abstractrunner.h   | 2 --
 2 files changed, 4 deletions(-)

diff --git a/src/abstractrunner.cpp b/src/abstractrunner.cpp
index 378c73d..34c44d1 100644
--- a/src/abstractrunner.cpp
+++ b/src/abstractrunner.cpp
@@ -244,12 +244,10 @@ void AbstractRunner::setHasRunOptions(bool hasRunOptions)
 }
 #endif
 
-#if KRUNNER_BUILD_DEPRECATED_SINCE(5, 71)
 void AbstractRunner::createRunOptions(QWidget *parent)
 {
     Q_UNUSED(parent)
 }
-#endif
 
 AbstractRunner::Speed AbstractRunner::speed() const
 {
diff --git a/src/abstractrunner.h b/src/abstractrunner.h
index cb4e32a..ae9558d 100644
--- a/src/abstractrunner.h
+++ b/src/abstractrunner.h
@@ -161,7 +161,6 @@ class KRUNNER_EXPORT AbstractRunner : public QObject
         bool hasRunOptions();
 #endif
 
-#if KRUNNER_ENABLE_DEPRECATED_SINCE(5, 71)
         /**
          * If hasRunOptions() returns true, this method may be called to get
          * a widget displaying the options the user can interact with to modify
@@ -172,7 +171,6 @@ class KRUNNER_EXPORT AbstractRunner : public QObject
          */
         KRUNNER_DEPRECATED_VERSION_BELATED(5, 71,  5, 0, "No longer use, feature removed")
         virtual void createRunOptions(QWidget *widget);
-#endif
 
         /**
          * Called whenever an exact or possible match associated with this
-- 
GitLab

